tests/system/providers/google/cloud/dataflow/example_dataflow_native_java.py

```diff
@@ -52,9 +52,14 @@
 PUBLIC_BUCKET = "airflow-system-tests-resources"
 
 JAR_FILE_NAME = "word-count-beam-bundled-0.1.jar"
+# SHARED_DATA_FOLDER should be set for Airflow which is running on distributed system.
+# For example in Composer the correct path is gcs/data/word-count-beam-bundled-0.1.jar.
+# Because gcs/data/ is shared folder for Airflow's workers.
+LOCAL_JAR_FILE_PATH = os.environ.get("SHARED_DATA_FOLDER")
```

Comment:
If we introduce a new environment variable, then we have to initiate it in our CI. Although we have scripts that initiate GCP projects and create Composer environments, adding anything test-specific into those high-level scripts isn't a good thing to do.

However, if we need to understand whether the test is running locally or in Composer, we actually could use Composer's [reserved environment variables](https://cloud.google.com/composer/docs/composer-2/set-environment-variables) that are always initiated in Composer.

For this purpose I'd suggest using `COMPOSER_ENVIRONMENT` in this way:
```python
IS_COMPOSER = bool(os.environ.get("COMPOSER_ENVIRONMENT", ""))
LOCAL_JAR_FILE_PATH = "gcs/data" if IS_COMPOSER else ""
```
I used this approach in the system test `cloudsql_query` already.

Reply:
I changed the PR description and also changed the `if` check what @moiseenkov suggested. 

@MaksYermak as we discussed in the daily, the Vertex AI's system test will change by @VladaZakharova.